src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -708,6 +708,98 @@ OverlayImpl::reportTraffic (
     m_traffic.addCount (cat, isInbound, number);
 }
 
+Json::Value
+OverlayImpl::crawlShards(bool pubKey, std::uint32_t hops)
+{
+    using namespace std::chrono;
+    using namespace std::chrono_literals;
+
+    Json::Value jv(Json::objectValue);
+    auto const numPeers {size()};
+    if (numPeers == 0)
+        return jv;
+
+    // If greater than a hop away, we may need to gather or freshen data
+    if (hops > 0)
+    {
+        // Prevent crawl spamming
+        clock_type::time_point const last(csLast_.load());
+        if (duration_cast<seconds>(clock_type::now() - last) > 60s)
```

Comment:
Is the `duration_cast` needed here? I think that @HowardHinnant's standardese magic ensures that `chrono` will do the right thing even in the absence of the cast.

Reply:
The compilers are happy without it and I am all for less.